server/backend/src/cameras/cameras.service.ts

```diff
@@ -1,27 +1,79 @@
-import { Injectable } from '@nestjs/common';
-import { EntityRepository } from '@mikro-orm/core';
+import { Injectable, Module } from '@nestjs/common';
+import { ConfigModule } from '@nestjs/config';
+import { Collection, EntityRepository } from '@mikro-orm/core';
 import { InjectRepository } from '@mikro-orm/nestjs';
+import { NotFoundError } from '../errors.types';
 import { Notification } from './notification.entity';
 import { Camera } from './camera.entity';
-import { NotFoundError } from '../errors.types';
+import { Expo } from 'expo-server-sdk';
+import * as dotenv from 'dotenv';
+dotenv.config();
 
 @Injectable()
 export class CamerasService {
   constructor(
-    @InjectRepository(Notification)
-    private notificationRepository: EntityRepository<Notification>,
     @InjectRepository(Camera)
     private cameraRepository: EntityRepository<Camera>,
+    @InjectRepository(Notification)
+    private notificationRepository: EntityRepository<Notification>,
   ) {}
 
-  sendNotification(id: string): boolean {
-    /* TODO: verify that ID refers to a valid camera and add a notification to DB */
-    return false;
+  /**
+   * Sends a notification to the user's phone and adds it to the
+   * collection of notifications the user has.
+   * @param idCam
+   */
+  public async sendNotification(idCam: string): Promise<boolean> {
+    const cam = await this.cameraRepository.findOne(
+      { id: idCam },
+      { populate: ['group.user'] },
+    );
+
+    if (cam === null) {
+      throw new NotFoundError(`Camera with given ID does not exist.`);
+    }
+    const expoClient = new Expo({ accessToken: process.env.EXPO_ACCESS_TOKEN });
+
+    const messages = [];
+    const expoToken = cam.group.user.expoToken;
+
+    if (!Expo.isExpoPushToken(expoToken)) {
+      console.error(`Push token ${expoToken} is not a valid Expo push token`);
+    }
+
+    messages.push({
+      to: expoToken,
+      sound: 'default',
+      title: `${cam.name} has detected movement!`,
+      body: 'This is a test notification',
+      data: { withSome: 'data' },
+    });
+    try {
+      const sent = await expoClient.sendPushNotificationsAsync(messages);
+    } catch (error) {
+      console.error(error);
+    }
```

Comment:
Rather than directly creating an Expo client and calling Expo SDK functions, we should have a generic `PushNotificationService` interface and an `ExpoPushNotificationService` implementation that we give to the `CamerasService`. That would help shrink this function down a bit and it would help if we need to switch from Expo later. See [here](https://github.com/tucker-moore/human-detector/blob/bbb50499dee5955dfbbeb2fec0df4db8778cd6aa/server/backend/src/auth/auth.module.ts#L7-L11), [here](https://github.com/tucker-moore/human-detector/blob/bbb50499dee5955dfbbeb2fec0df4db8778cd6aa/server/backend/src/auth/auth-service.interface.ts) and [here](https://github.com/tucker-moore/human-detector/blob/bbb50499dee5955dfbbeb2fec0df4db8778cd6aa/server/backend/src/auth/jwt-identity.guard.ts#L19) for an example of defining and injecting an interface implementation (note that the KeycloakAuthService file is weird, you can just define a class that implements your interface rather than a function)